File: src/CabinConnect.Api/Domain/Cabin.cs
namespace CabinConnect.Api.Domain;

public sealed record Cabin(
    Guid Id,
    Guid HostId,
    string Name,
    string Description,
    string StreetAddress,
    int Capacity,
    bool IsActive,
    IReadOnlyList<Amenity> CatalogAmenities,
    IReadOnlyList<string> CustomAmenities);

File: web/src/types/cabin.ts
export interface AmenityResponse {
  id: string;
  displayName: string;
}

export interface CabinResponse {
  id: string;
  name: string;
  description: string;
  streetAddress: string;
  capacity: number;
  isActive: boolean;
  catalogAmenities: AmenityResponse[];
  customAmenities: string[];
}

export interface RegisterCabinRequest {
  name: string;
  description: string;
  streetAddress: string;
  capacity: number;
  catalogAmenityIds: string[] | null;
  customAmenities: string[] | null;
}

export interface UpdateCabinRequest {
  name: string;
  description: string;
  streetAddress: string;
  capacity: number;
  catalogAmenityIds: string[] | null;
  customAmenities: string[] | null;
}
